onboarding/main.go

```diff
@@ -0,0 +1,161 @@
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	onboardingTicketPublicKeySecret  = "onboarding-ticket-public-key"
+	onboardingTicketPrivateKeySecret = "onboarding-ticket-private-key"
+)
+
+var (
+	ctx                   = context.Background()
+	serverNamespace       = os.Getenv(util.WatchNamespaceEnvVar)
+	privatePem, publicPem string
+)
+
+func createSecret(cl client.Client, keyName, secretName string) {
+
+	desiredSecret := GetProviderAPIServerSecret(keyName, secretName)
+	actualSecret := &corev1.Secret{}
+
+	err := cl.Get(ctx, client.ObjectKeyFromObject(desiredSecret), actualSecret)
+
+	if err != nil && errruntime.IsNotFound(err) {
+		err = cl.Create(ctx, desiredSecret)
+		if err != nil {
+			klog.Error(err, "Failed to create secret.")
+		}
+	} else if err != nil {
+		klog.Error(err, "Failed to get secret.")
+
+	}
+}
+
+func GetProviderAPIServerSecret(key string, secretName string) *corev1.Secret {
+
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: serverNamespace,
+		},
+		Immutable: func(flag bool) *bool { return &flag }(true),
+		StringData: map[string]string{
+			"data": key,
+		},
+		// Type: v1.SecretTypeServiceAccountToken,
+	}
+}
+
+func newClient() (client.Client, error) {
+	scheme := apiruntime.NewScheme()
+	utilruntime.Must(ocsv1.AddToScheme(scheme))
+	utilruntime.Must(corev1.AddToScheme(scheme))
+	utilruntime.Must(ocsv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(openshiftv1.AddToScheme(scheme))
+
+	config, err := config.GetConfig()
+	if err != nil {
+		klog.Error(err, "failed to get config check if KUBECOFIG is set.")
+		return nil, err
+	}
+	client, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		klog.Error(err, "failed to create controller-runtime client")
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func ConvertRsaPrivateKeyAsPemStr(privateKey *rsa.PrivateKey) string {
+	privteKeyBytes := x509.MarshalPKCS1PrivateKey(privateKey)
+	privateKeyPem := pem.EncodeToMemory(
+		&pem.Block{
+			Type:  "RSA PRIVATE KEY",
+			Bytes: privteKeyBytes,
+		},
+	)
+	return string(privateKeyPem)
+}
+
+func ConvertRsaPublicKeyAsPemStr(publicKey *rsa.PublicKey) (string, error) {
+	publicKeyBytes, err := x509.MarshalPKIXPublicKey(publicKey)
+	if err != nil {
+		return "", err
+	}
+	publicKeyPem := pem.EncodeToMemory(
+		&pem.Block{
+			Type:  "RSA PUBLIC KEY",
+			Bytes: publicKeyBytes,
+		},
+	)
+
+	return string(publicKeyPem), nil
+}
+
+func genratePrivateAndPublicKeys() {
+
+	// Generate RSA key.
+	var err error
+	privateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		panic(err)
+	}
+
+	publicKey := &privateKey.PublicKey
+
+	// Export the keys to pem string
+	//similar to key.pem & pubkey.pem
+	privatePem = ConvertRsaPrivateKeyAsPemStr(privateKey)
+	publicPem, _ = ConvertRsaPublicKeyAsPemStr(publicKey)
```

Comment:
Why are we ignoring the error?

Reply:
It's not required as it will convert all the time. I am adding it now for the safer side.